<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PlayVideoUI</class>
 <widget class="QMainWindow" name="PlayVideoUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>视频流客户端</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainLayout">
    <item>
     <widget class="QStackedWidget" name="mainStackedWidget">
      <widget class="QWidget" name="videoListPage">
       <layout class="QVBoxLayout" name="listPageLayout">
        <item>
         <widget class="QGroupBox" name="serverGroupBox">
          <property name="title">
           <string>服务器连接</string>
          </property>
          <layout class="QHBoxLayout" name="serverLayout">
           <item>
            <widget class="QLabel" name="serverLabel">
             <property name="text">
              <string>服务器地址:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="serverInput">
             <property name="text">
              <string>http://localhost:5000</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="connectButton">
             <property name="text">
              <string>连接</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="serverHorizontalSpacer">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="uploadGroupBox">
          <property name="title">
           <string>上传视频</string>
          </property>
          <layout class="QHBoxLayout" name="uploadLayout">
           <item>
            <widget class="QLineEdit" name="selectedVideoPath">
             <property name="readOnly">
              <bool>true</bool>
             </property>
             <property name="placeholderText">
              <string>请选择视频文件...</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="browseButton">
             <property name="text">
              <string>浏览</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="uploadButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>上传</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QScrollArea" name="videoListScrollArea">
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarPolicy::ScrollBarAsNeeded</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="videoListContentWidget">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>946</width>
             <height>410</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="videoListLayout">
            <property name="leftMargin">
             <number>10</number>
            </property>
            <property name="topMargin">
             <number>10</number>
            </property>
            <property name="rightMargin">
             <number>10</number>
            </property>
            <property name="bottomMargin">
             <number>10</number>
            </property>
            <property name="spacing">
             <number>10</number>
            </property>
            <property name="sizeConstraint">
             <enum>QLayout::SetMinAndMaxSize</enum>
            </property>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="statusLayout">
          <item>
           <widget class="QLabel" name="statusLabel">
            <property name="text">
             <string>准备连接服务器...</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="refreshButton">
            <property name="text">
             <string>刷新</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="progressBar">
            <property name="value">
             <number>0</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="videoPlayerPage">
       <layout class="QVBoxLayout" name="playerPageLayout">
        <item>
         <layout class="QHBoxLayout" name="topToolBarLayout">
          <item>
           <widget class="QToolButton" name="returnButton">
            <property name="text">
             <string>← 返回</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="topRightSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="downloadButton">
            <property name="text">
             <string>下载</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QVideoWidget" name="videoWidget" native="true">
          <property name="minimumSize">
           <size>
            <width>640</width>
            <height>480</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="controlLayout">
          <item>
           <widget class="QPushButton" name="playButton">
            <property name="text">
             <string>播放</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pauseButton">
            <property name="text">
             <string>暂停</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="stopButton">
            <property name="text">
             <string>停止</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="controlHorizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="volumeLabel">
            <property name="text">
             <string>音量:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="volumeSlider">
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>50</number>
            </property>
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="controlRightSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="progressLayout">
          <item>
           <widget class="QLabel" name="currentTimeLabel">
            <property name="text">
             <string>00:00</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="progressSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="totalTimeLabel">
            <property name="text">
             <string>00:00</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>

       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVideoWidget</class>
   <extends>QWidget</extends>
   <header>qvideowidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
